drivers: serial: nrfx_uarte: fix deprecation warning when UART unused

xudongzheng opened the pull request:
This hides the deprecation warning when UART_NRFX_UARTE_LEGACY_SHIM is false due to UART being unused.

Warning was introduced in #81536.

xudongzheng committed:
drivers: serial: nrfx_uarte: fix deprecation warning when UART is unused

This hides the deprecation warning when UART_NRFX_UARTE_LEGACY_SHIM is
false because UART is not used.

Signed-off-by: Xudong Zheng <7pkvm5aw@slicealias.com>

nordic-krch commented on soc/nordic/Kconfig:
```diff
@@ -199,7 +199,7 @@ config NRF_PLATFORM_HALTIUM
 config DEPRECATED_UART_NRFX_UARTE_LEGACY_SHIM
 	bool
-	default y if !UART_NRFX_UARTE_LEGACY_SHIM
+	default y if !UART_NRFX_UARTE_LEGACY_SHIM && UART_NRFX_UARTE && UART_NRFX && SERIAL
```
isn't that enough?
```suggestion
	default y if !UART_NRFX_UARTE_LEGACY_SHIM && UART_NRFX_UARTE
```

xudongzheng replied:
Both should be addressed in latest version.

xudongzheng committed:
drivers: serial: nrfx_uarte: move DEPRECATED_UART_NRFX_UARTE_LEGACY_SHIM

The Kconfig option should be defined alongside
UART_NRFX_UARTE_LEGACY_SHIM for clarity.

Signed-off-by: Xudong Zheng <7pkvm5aw@slicealias.com>
